src/main/java/io/debezium/connector/spanner/db/model/schema/Column.java

```diff
@@ -44,46 +47,103 @@ public boolean isNullable() {
         spannerType = spannerType.toUpperCase();
-        if ("BOOL".equals(spannerType)) {
-            return new ColumnType(DataType.BOOL);
-        }
-        if ("INT64".equals(spannerType)) {
-            return new ColumnType(DataType.INT64);
-        }
-        if ("FLOAT64".equals(spannerType)) {
-            return new ColumnType(DataType.FLOAT64);
-        }
-        if (spannerType.startsWith("STRING")) {
-            return new ColumnType(DataType.STRING);
-        }
-        if (spannerType.startsWith("BYTES")) {
-            return new ColumnType(DataType.BYTES);
-        }
-        if ("TIMESTAMP".equals(spannerType)) {
-            return new ColumnType(DataType.TIMESTAMP);
+        switch (dialect) {
+            case GOOGLE_STANDARD_SQL:
+                if ("BOOL".equals(spannerType)) {
+                    return new ColumnType(DataType.BOOL);
+                }
+                if ("INT64".equals(spannerType)) {
+                    return new ColumnType(DataType.INT64);
+                }
+                if ("FLOAT64".equals(spannerType)) {
+                    return new ColumnType(DataType.FLOAT64);
+                }
+                if (spannerType.startsWith("STRING")) {
+                    return new ColumnType(DataType.STRING);
+                }
+                if (spannerType.startsWith("BYTES")) {
+                    return new ColumnType(DataType.BYTES);
+                }
+                if ("TIMESTAMP".equals(spannerType)) {
+                    return new ColumnType(DataType.TIMESTAMP);
+                }
+                if ("DATE".equals(spannerType)) {
+                    return new ColumnType(DataType.DATE);
+                }
+                if ("NUMERIC".equals(spannerType)) {
+                    return new ColumnType(DataType.NUMERIC);
+                }
+                if ("JSON".equals(spannerType)) {
+                    return new ColumnType(DataType.JSON);
+                }
+                if (spannerType.startsWith("ARRAY")) {
+                    // Substring "ARRAY<xxx>"
+                    String spannerArrayType = spannerType.substring(6, spannerType.length() - 1);
+                    ColumnType itemType = parseColumnType(spannerArrayType, dialect);
+                    return new ColumnType(DataType.ARRAY, itemType);
+                }
+                throw new IllegalArgumentException("Unknown spanner type " + spannerType);
+            case POSTGRESQL:
+                if (spannerType.endsWith("[]")) {
+                    // Substring "xxx[]"
+                    // Must check array type first
+                    String spannerArrayType = spannerType.substring(0, spannerType.length() - 2);
+                    ColumnType itemType = parseColumnType(spannerArrayType, dialect);
+                    return new ColumnType(DataType.ARRAY, itemType);
+                }
+                if ("BOOLEAN".equals(spannerType)) {
```

Comment:
I don't see the JsonB data type or the spanner commit timestamp data type.

In addition, do we want to remove the Spanner type length limits?

Reply:
Added SPANNER.COMMIT_TIMESTAMP, Referred this file for relevant changes http://shortn/_JT4iJYizUg, but did not have JsonB in it. Spanner type length limits only apply for arrays and they were already in place before and also exist in the referred file above.